dali/image/png.cc

```diff
@@ -16,12 +16,94 @@
 
 namespace dali {
 
-// Assume chunk points to a 4-byte value
-int ReadIntFromPNG(const uint8 *chunk) {
-  // reverse the bytes, cast
-  return (unsigned int) (chunk[0] << 24 | chunk[1] << 16 | chunk[2] << 8 | chunk[3]);
+namespace detail {
+
+  // https://www.w3.org/TR/PNG-Chunks.html
+
+  // Layout:
+  // 4 bytes: chunk size (should be 13 bytes for IHDR)
+  // 4 bytes: Chunk Identifier (should be "IHDR")
+  // 4 bytes: Width
+  // 4 bytes: Height
+  // 1 byte : Bit Depth
+  // 1 byte : Color Type
+  // 1 byte : Compression method
+  // 1 byte : Filter method
+  // 1 byte : Interlace method
+
+enum {
+  SIZE_CHUNK_SIZE = 4,
```

Comment:
Could you use `cosntexpr` variables instead of enums?

Reply:
done